pkg/utils/test_utils.go

```diff
@@ -0,0 +1,25 @@
+package utils
+
+import (
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func CreateNodesList(numNodes int, labels map[string]string) *corev1.NodeList {
+	nodesList := corev1.NodeList{}
+	nodesList.Items = make([]corev1.Node, numNodes)
+	for i := range nodesList.Items {
+		nodesList.Items[i].APIVersion = "v1"
+		nodesList.Items[i].Kind = "Node"
+		nodesList.Items[i].Name = fmt.Sprintf("Node%d", i)
+		nodesList.Items[i].SetLabels(labels)
+	}
+
+	return &nodesList
+}
+
+func AddTaint(node *corev1.Node, taintKey, taintValue string, taintEffect corev1.TaintEffect) {
+	var newTaints []corev1.Taint
+	newTaints = append(newTaints, corev1.Taint{Key: taintKey, Value: taintValue, Effect: taintEffect})
+	node.Spec.Taints = newTaints
```

Comment:
I think this erases previous taints (not a big deal since this is for tests), but how about changing it to:
```suggestion
	node.Spec.Taints = append(
		node.Spec.Taints,
		corev1.Taint{Key: taintKey, Value: taintValue, Effect: taintEffect},
	}
```

Reply:
since this is a test, and we create the nodes struct, they don't actually have any taints defined at any point

Reply:
in order to take into account existing taints, we need to check if taints exist, and if a new taint is already defined (key,value, effect), and add it only in case there is a difference. Currently fill like an overkill for unit-test puproses

Reply:
Fair enough. Then how about renaming the function to `SetTaint`?
```suggestion
	node.Spec.Taints = []corev1.Taint{
		{Key: taintKey, Value: taintValue, Effect: taintEffect},
	}
```

Reply:
fixed